swift/IceDiscovery/helloUI/Client.swift

```diff
@@ -0,0 +1,199 @@
+                queuedRequest("hello")
+            }
+        } catch {
+            exception(error)
+        }
+    }
+
+    func flushBatch() {
+        if helloPrx == nil {
+            return
+        }
+        firstly {
+            helloPrx.ice_flushBatchRequestsAsync()
+        }.done {
+            self.flushBatchSend()
+        }.catch { error in
+            self.exception(error)
+        }
+    }
+
+    func shutdown() {
+        do {
+            if helloPrx == nil {
+                return
+            }
+
+            if proxySettings.deliveryMode != .OnewayBatch,
+               proxySettings.deliveryMode != .OnewaySecureBatch,
+               proxySettings.deliveryMode != .DatagramBatch
+            {
+                sendingRequest()
+                var response = false
+                firstly {
+                    helloPrx.shutdownAsync { _ in
+                        if !response {
+                            self.requestSent()
+                        }
+                    }
+                }.done {
+                    response = true
+                    self.configure(comm: self.communicator)
+                    self.ready()
+                }.catch { error in
+                    response = true
+                    self.exception(error)
+                    self.ready()
+                }
+            } else {
+                try helloPrx.shutdown()
+                queuedRequest("shutdown")
+            }
+        } catch {
+            exception(error)
+            ready()
+        }
+    }
+
+    func updateProxy() throws {
+        let mode = proxySettings.deliveryMode
+        helloPrx = uncheckedCast(prx: try communicator.propertyToProxy("HelloProxy")!, type: HelloPrx.self)
+
+        switch mode {
+        case .Twoway:
+            helloPrx = helloPrx.ice_twoway()
+        case .TwowaySecure:
+            helloPrx = helloPrx.ice_twoway().ice_secure(true)
+        case .ModeOneway:
+            helloPrx = helloPrx.ice_oneway()
+        case .OnewayBatch:
+            helloPrx = helloPrx.ice_batchOneway()
+        case .OnewaySecure:
+            helloPrx = helloPrx.ice_oneway().ice_secure(true)
+        case .OnewaySecureBatch:
+            helloPrx = helloPrx.ice_batchOneway().ice_secure(true)
+        case .Datagram:
+            helloPrx = helloPrx.ice_datagram()
+        case .DatagramBatch:
+            helloPrx = helloPrx.ice_batchDatagram()
+        }
+
+        if proxySettings.timeout != 0 {
+            helloPrx = helloPrx.ice_invocationTimeout(Int32(proxySettings.timeout))
+        }
+    }
+
+    func sendingRequest() {
+        helloEnabled = false
+        shutdownEnabled = false
+        statusMessage = "Sending request"
+        isSpinning = true
+    }
+
+    func requestSent() {
+        if let connection = helloPrx.ice_getCachedConnection() {
+            do {
+                // Loop through the connection informations until we find an IPConnectionInfo class.
```

Comment:
In this discovery demo, hostname is read-only and we show the hostname (but not port number) of the connection used to send the request. 

I think we could do clearer, and just print the connection (and rename hostname to connection) to show which connection was used to send the request.